mlrun/projects/project.py

```diff
@@ -294,6 +294,11 @@ def load_project(
 
     if save and mlrun.mlconf.dbpath:
         project.save()
+        # add secrets to the saved project before registering artifacts because they might be needed
+        # to download the artifacts from external sources
+        if secrets:
+            project.set_secrets(secrets=secrets)
```

Comment:
I understand from @GiladShapira94 that this part is controversial.
I need secrets to get the artifacts from s3 but if we want secrets to load and project secrets to set we can add a separate parameter